tests/components/notify/test_init.py

```diff
@@ -97,3 +123,52 @@ async def test_warn_template(hass, caplog):
     # We should only log it once
     assert caplog.text.count("Passing templates to notify service is deprecated") == 1
     assert hass.states.get("persistent_notification.notification") is not None
+
+
+async def test_setup_platform_and_reload(hass, caplog, tmp_path):
+    """Test service setup and reload."""
+    get_service_called = Mock()
+
+    async def async_get_service(hass, config, discovery_info=None):
+        """Get notify service for mocked platform."""
+        get_service_called()
+        targetlist = {"a": 1, "b": 2}
+        return NotificationService(hass, targetlist)
+
+    # Mock notify services file
+    services_yaml_file = tmp_path / "services.yaml"
+    services_config = yaml.dump(
+        {"notify": [{"description": "My custom notify service", "fields": {}}]}
+    )
+    services_yaml_file.write_text(services_config)
+    loaded_platform = MockNotifyPlatform(async_get_service=async_get_service)
+    mock_platform(hass, "testnotify.notify", loaded_platform)
+    integration = hass.data[DATA_INTEGRATIONS]["testnotify"]
+    integration.file_path = tmp_path
+
+    # Setup the platform
+    await notify.async_setup(hass, {"notify": [{"platform": "testnotify"}]})
```

Comment:
Please use `async_setup_component` instead.

Reply:
`await async_setup_component(hass, "testnotify", {})` does not seem to initialize `notify.async_setup`.

Reply:
The notify integration needs to be setup. Either as a dependency of another integration or explicitly.

Reply:
If I add  `await async_setup_component(hass, "notify", {})` This still does not initialize. The integration platform is loaded:
```
INFO     homeassistant.loader:loader.py:415 Loaded notify from homeassistant.components.notify
INFO     homeassistant.setup:setup.py:204 Setting up notify
DEBUG    homeassistant.core:core.py:847 Bus:Handling <Event service_registered[L]: domain=notify, service=persistent_notification>
INFO     homeassistant.setup:setup.py:252 Setup of domain notify took 0.0 seconds
DEBUG    homeassistant.core:core.py:847 Bus:Handling <Event component_loaded[L]: component=notify>
INFO     homeassistant.loader:loader.py:415 Loaded testnotify from homeassistant.components.testnotify
INFO     tests.common:common.py:1155 Adding mock integration: testnotify
INFO     tests.common:common.py:1184 Adding mock integration platform: testnotify.notify
INFO     homeassistant.setup:setup.py:204 Setting up testnotify
INFO     homeassistant.setup:setup.py:252 Setup of domain testnotify took 0.0 seconds
DEBUG    homeassistant.core:core.py:847 Bus:Handling <Event component_loaded[L]: component=testnotify>
```

Reply:
If we want to set up the platform from config, we need to add the platform section in the config under the `"notify"` domain key in the call to `async_setup_component`.
Eg:

```py
await async_setup_component(hass, "notify", {"notify": {"platform": "testnotify"}})
await hass.async_block_till_done()

Reply:
Thnx, that did the trick